nexus_keycode/protocols/utils.py

```diff
@@ -58,3 +58,13 @@ def chunk(iteration):
     output_bits = bitstring.Bits().join(chunk(i) for i in range(iterations))
 
     return output_bits[:output_len]
+
+
+def generate_mac(input_val, secret_key):
+    """
+    :type input_val: 'byte'
+    :type secret_key: 'byte'
+    """
+    # Return the last 6 digits of the siphash result
+    function = siphash.SipHash_2_4(secret_key, input_val)
+    return u"{:06d}".format(function.hash() & 0xFFFFFFFF)[-6:]
```

Comment:
masks lower 32 bits, and returns 6 digits :+1:

Reply:
So, this is actually a `full` protocol MAC, it is not the same MAC we use for all protocols here (for example, `small` protocol does not use this MAC).

This makes me wonder whether this belongs in the `utils` folder, but I think its fine since we need it for a separate use case here. Can we name it something a bit more descriptive, e.g. "generate_base10_mac" or something?